### PR TITLE
Adapt users_finish client

### PR DESCRIPTION
**Pull request by joseivanlopez**
### Problem

Clients *inst_user_first* and *inst_root_first* were already adapted to use the new `Y2Users` classes, but *users_finish* client is still using the old code to create the users into the system.

### Solution

The *users_finish* client was adapted to use `Y2Users` classes.

NOTE: *users_finish* is also used by AutoYaST, but the AutoYaST part will be adapted in a separate PR.

## Testing

* Added unit tests
* Manually tested:
  * Autologin is not working (will be fixed in a separate PR)
  * Root password form has an issue: https://github.com/yast/yast-users/pull/273

### CODE REVIEW

File: test/users_finish_test.rb [hunk 1]
```diff
@@ -71,29 +74,87 @@
...
+          it "reports all errors to the user" do
+            expect(Yast::Report).to receive(:Error).with("error 1\n\nerror 2")
+
+            subject.run
+          end
         end
 
-        context "when a new user is added" do
-          let(:user_added) { true }
+        context "when an user is configured for auto-login" do
```
Comment (dgdavid) [inline test/users_finish_test.rb:138]: NP: I think it is [_**a** user_](https://english.stackexchange.com/questions/105116/is-it-a-user-or-an-user). And I think now that probably I wrote _an user_ in the past and/or did not catch them in previous reviews.

File: src/lib/users/clients/users_finish.rb [hunk 2]
```diff
@@ -82,12 +93,86 @@ def write_autoinst
       Progress.set(@progress_orig)
     end
 
-    # Write root password a new users during regular installation
+    # Writes users during the installation
+    #
+    # The linux writer will proceed all the differences between the system and the target configs.
```
Comment (ancorgs) [inline src/lib/users/clients/users_finish.rb:98]: proceed -> process

File: src/lib/users/clients/users_finish.rb [hunk 3]
```diff
@@ -82,12 +93,86 @@ def write_autoinst
...
+    def system_config
+      config_manager.system
+    end
+
+    # Target config, which extends the system config with the new users that should be created
+    # during the installation.
+    #
+    # @return [Y2Users::Config]
+    def target_config
+      target = config_manager.config(:target)
```
[the remark this answers is not in the document]
Comment (joseivanlopez) [reply src/lib/users/clients/users_finish.rb:141]: No real benefit, but as we need a ConfigMaganer in order to retrieve system ... But we can avoid to store *target* there.

File: src/lib/users/clients/users_finish.rb [hunk 4]
```diff
@@ -82,12 +93,86 @@ def write_autoinst
...
+
+      Autologin.user = UsersSimple.GetAutologinUser
+      Autologin.Use(true)
+    end
+
+    # System config, which contains all the current users on the system
+    #
+    # @return [Y2Users::Config]
+    def system_config
+      config_manager.system
```
Comment (ancorgs) [inline src/lib/users/clients/users_finish.rb:133]: I wonder if we should use `force_read` here.

File: src/lib/users/clients/users_finish.rb [hunk 5]
```diff
@@ -82,12 +93,86 @@ def write_autoinst
...
+      Y2Users::ConfigManager.instance
+    end
+
+    # Reports issues
+    #
+    # @param issues [Array<Y2Issues::Issue>]
+    def report_issues(issues)
+      message = issues.map(&:message).join("\n\n")
+
+      Report.Error(message)
```
[the remark this answers is not in the document]
Comment (joseivanlopez) [reply src/lib/users/clients/users_finish.rb:175]: Created https://trello.com/c/08t84wmL/4716-a-better-way-for-reporting-issues-during-installation.

**Review by ancorgs — APPROVED** [review]
LGTM

**Pull request merged**